api/src/main/scala/za/co/absa/loginsvc/rest/service/AuthSearchService.scala

```diff
@@ -0,0 +1,139 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.rest.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import za.co.absa.loginsvc.model.User
+import za.co.absa.loginsvc.rest.config.auth.{ActiveDirectoryLDAPConfig, UsersConfig}
+import za.co.absa.loginsvc.rest.config.provider.AuthConfigProvider
+import za.co.absa.loginsvc.rest.config.validation.ConfigValidationException
+
+import java.util
+import javax.naming.Context
+import javax.naming.directory.{Attributes, DirContext, SearchControls, SearchResult}
+import javax.naming.ldap.{Control, InitialLdapContext, PagedResultsControl}
+import scala.collection.JavaConverters.enumerationAsScalaIteratorConverter
+
+@Service
+class AuthSearchService @Autowired()(authConfigProvider: AuthConfigProvider) {
+
+  private val logger = LoggerFactory.getLogger(classOf[AuthSearchService])
+
+  private val usersConfig = authConfigProvider.getUsersConfig
+  private val adLDAPConfig = authConfigProvider.getLdapConfig
+
+  private val configs = Array(usersConfig, adLDAPConfig).filter(_.order != 0).sortBy(_.order)
+  if (configs.isEmpty)
+    throw ConfigValidationException("No authentication method enabled in config")
+
+  def searchUser(username: String): User = {
+
+    configs.foreach {
+      case u: UsersConfig =>
+        val result = searchUsersConfig(username)
+        if (result.isDefined) return result.get
+      case l: ActiveDirectoryLDAPConfig =>
+        val result = searchLDAP(username)
+        if (result.isDefined) return result.get
+      case _ => //ignore for now
+    }
+    throw new NoSuchElementException(s"Value not found in any object.")
+  }
+
+  private def searchUsersConfig(username: String): Option[User] = {
+    usersConfig.knownUsersMap.get(username).flatMap { userConfig =>
+      val optionalAttributes: Map[String, Option[AnyRef]] = userConfig.attributes.getOrElse(Map.empty).map {
+        case (k, v) => (k, Some(v))
+      }
+      Some(User(username, userConfig.groups.toList, optionalAttributes))
+    }
+  }
+
+  private def searchLDAP(username: String): Option[User] = {
+    val serviceAccount = adLDAPConfig.serviceAccount
+    val context = getDirContext(serviceAccount.username, serviceAccount.password)
+      try {
+        val users = context
+          .search(adLDAPConfig.domain.split("\\.").map(part => s"dc=$part").mkString(","),
+            adLDAPConfig.searchFilter.replace("{1}", username),
+            getSimpleSearchControls)
+          .asScala.filter(filterSearchResults).map(resultToUserEntry).toList
+
+        if (users.nonEmpty) {
+          println("User found in LDAP")
```

Comment:
```suggestion
```
remove `println`s in the end or replace with logging. (here, probably remove, seems like a dev/debug residuum)

Reply:
Yeah, sorry, this is left over from debugging